src/utils.ts

```diff
@@ -2,25 +2,29 @@ import fs from 'fs';
 import { getPullRequestNumber, getTagCommitSha, isMainBranch, isChangeInPath } from './github';
 import { debug } from '@actions/core';
 
-export const getPackagesToDeploy = (
+export const getChangedPackages = (
   lastDeployedRef: string,
   rushProjects: RushProjects[]
 ): ActionOutputs => {
-  return rushProjects.reduce<ActionOutputs>(
-    (output, project) => {
-      if (isChangeInPath(lastDeployedRef, project.projectFolder)) {
-        const deployCategory = getDeployCategory(project.projectFolder);
-        if (deployCategory) {
-          output[deployCategory].push(project.projectFolder);
-        }
+  return rushProjects.reduce<ActionOutputs>((output, project) => {
```

Comment:
I think reduce gets quite hard to read, and breaking the callback out into a named function often helps. The function can still be declared inside of the main function to give it access to the same variables.